### PR TITLE
Support running node_interop in null safety mode

### PR DESCRIPTION
**Pull request by Awjin**
This allows downstream users to start migrating their packages to null safety.

### CODE REVIEW

File: node_interop/pubspec.yaml
Comment (pulyaevskiy) [inline node_interop/pubspec.yaml]: According to [migration guide](https://dart.dev/null-safety/migration-guide) we should bump this to `2.0.0-nullsafety.0`.

**Review by pulyaevskiy — CHANGES_REQUESTED** [review]
Thanks! LGTM, just a minor comment in pubspec. Please take a look.

File: node_interop/pubspec.yaml
Comment (Awjin) [reply node_interop/pubspec.yaml]: Good catch, thanks. What do you think about bumping the minor version instead? Bumping major seems overly conservative given that consumers won't be using the non-nullable API before they upgrade to Dart 2.12 anyways. Your call

Comment (pulyaevskiy) [reply node_interop/pubspec.yaml]: We could go with `1.3.0` though I generally prefer to stick to guidelines to avoid confusion, be consistent and promote best practices. Also I feel like bumping major version is justified here by the fact that we depend on an entirely new language feature. Versioning logistics are also a bit simpler: if a consumer uses Dart 2.11 or older they just need to stick to the 1.x branch, but for everyone on 2.12+ it's 2.x. Don't need to look for which particular 1.x version supports null safety and which is not.

**Comment by Awjin** [comment]
Looks like this is breaking the e2e test that invokes `build_runner` (which does not have a null-safe version available).

**Comment by pulyaevskiy** [comment]
> Looks like this is breaking the e2e test that invokes `build_runner` (which does not have a null-safe version available).

@Awjin 

Build on stable fails because it runs on Dart 2.10.x and we've updated SDK constraint here.
Build on dev fails for multiple reasons. There seems to be an issue with build_node_compilers as well as in node_interop. I suspect e2e test fails because of the first one.

The node_interop one seems to be easy to fix, it's just a lint:
https://travis-ci.org/github/pulyaevskiy/node-interop/jobs/748668797#L557

We may need to look at fixing build_node_compilers first before landing this change...

**Comment by jathak** [comment]
`build_node_compilers` probably can't be migrated to null-safety until `build` (and other related packages) are, and it's [blocked on `analyzer` being migrated](https://github.com/dart-lang/build/issues/2920#issuecomment-732230445), which is apparently not happening until [March at the earliest](https://github.com/dart-lang/sdk/issues/44290#issuecomment-733181379).

This shouldn't block migrating packages that only use `build_node_compilers` as a dev dependency though. Looking into this now...

**Comment by jathak** [comment]
Dart 2.12 was released as stable this morning, so I'm going to check again to see if we can migrate without waiting on `build`. If not, it hopefully won't be a blocker for long, since `analyzer`'s now been migrated and it looks like the `build` packages are being actively worked on.

**Comment by Awjin** [comment]
@pulyaevskiy @jathak I went through and fixed broken lints and test, then rebased the `node_interop` nnbd changes onto the fixes.

`e2e_interop_test` and `build_node_compilers` are still broken. Fixing `e2e_interop_test` depends on fixing `build_node_compilers`, which is in turn blocked on `build` migrating to nnbd.

**Comment by Awjin** [comment]
Ok, I've worked around all of the breakages. `build_node_compilers` works now—a deeper fix requires migrating it to nnbd, which is blocked on `build` migrating first.

@pulyaevskiy WDYT about releasing this first, then migrating `build_node_compilers` once it's possible to do so? This would unblock Dart Sass's migration to nnbd.

File: e2e_interop_test/build.yaml
Comment (pulyaevskiy) [inline e2e_interop_test/build.yaml:4]: Just сurious - is this a new addition in build package?

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks.

**Comment by pulyaevskiy** [comment]
@Awjin sorry for the delay here. 

> WDYT about releasing this first, then migrating build_node_compilers once it's possible to do so? This would unblock Dart Sass's migration to nnbd.

This makes sense to me. Will merge this and publish a new version.

Thanks for working on this!

File: e2e_interop_test/build.yaml
Comment (Awjin) [reply e2e_interop_test/build.yaml:4]: Yes, this is a new placeholder added in `build_runner`

**Comment by Awjin** [comment]
@pulyaevskiy Thanks for the review and merge! Would you be able to release a new version as well?

**Pull request merged**
